### DISCUSSION TITLE
add new lifecycle hook

### DISCUSSION THREAD

**Discussion by jantimon** [dotansimha/graphql-code-generator#8643] [Ideas] [answered]
hello 👋

I would like to propose an improved `beforeOneFileWrite` [lifecycle hook](https://the-guild.dev/graphql/codegen/docs/config-reference/lifecycle-hooks).  
It should allow to modify the generated content:

`(file:string, content: string) => Promise<string | null> | string | null` 

There are three advantages over the current `beforeOneFileWrite` behaviour:

1. it would allow to take prettification or linter fixes into account for the hash checksum for better checking if the file needs to be rewritten.
2. it would be faster as it does not require to startup node
3. it would trigger less file watcher notifications in watch mode

Would love to hear your feedback on the idea itself and maybe an even better name for such a hook.

Update:

I prepared a PR (#8662)  which would allow `beforeOneFileWrite` to modify the content.
It depends on 2 small refactoring PRS: #8652 & #8661 

Example Usage:

```js
hooks: {
    beforeOneFileWrite: async (_file, content) => prettier.format(content, {
      parser: "typescript"
    }),
}
```

**Comment by saihaj** [comment]
My opinion and what I recommend is not committing generated code so to me I am not really a fan of this hook

**Reply by jantimon** [reply]
I would prefer not to mix this discussion with best practices of git and deployment workflows.

Post generation modification API hooks already exist.

My proposal is to make those API hooks more efficient and performant - especially for watch mode usage.

**Comment by saihaj** [comment, marked as answer]
thanks to @jantimon added here https://github.com/dotansimha/graphql-code-generator/releases/tag/release-1670443964814
